src/js/webrequest.js

```diff
@@ -706,6 +715,7 @@ function startListeners() {
   chrome.webRequest.onHeadersReceived.addListener(onHeadersReceived, {urls: ["<all_urls>"]}, ["responseHeaders", "blocking"]);
   chrome.tabs.onRemoved.addListener(onTabRemoved);
   chrome.tabs.onReplaced.addListener(onTabReplaced);
+  chrome.tabs.onUpdated.addListener(onTabUpdated);
```

Comment:
Sure, I took a quick look and it doesn't seem to be too hard.

As far as pulling out the badge changes, it might be a bit tricky; looks like I'd have to pull out a number of non-linear commits. I do see your point though, as there seems to be enough work around the badge to warrant another PR, though I think it does still fit under the scope of performance improvements, since we're moving it to (a) only be called when new trackers are detected, and (b) be called in an async runtime.

I do have some thoughts about how the badge number is being counted. Currently, each time `updateBadge()` is called, we go through every third-party domain that has been loaded on the specified tab and find the ones with a stored action present in `BLOCKED_ACTIONS`. This is O(n^2) runtime, albeit with n being relatively small. I'm not sure whether improving this is a priority right now, though to improve it to linear runtime would require some changes to the `tabData` data structure that @cowlicks has been tweaking in another PR, so not entirely sure this PR would be the best place for that in any case.